packages/react/src/shared/proxy/utilsProxy.ts

```diff
@@ -146,6 +146,25 @@ type DecorateProcedure<
   ): InfiniteData<inferProcedureOutput<TProcedure>> | undefined;
 };
 
+/**
+ * this is the type that is used to add in procedures that can be used on
+ * partial paths.
+ *
+ * NOTE: All generics have been removed from this type at the moment ... But we
+ * may want to add these back in if we can be smarter about inferring input
+ * type...
+ */
+type DecoratePartialPathProcedure = {
+  /**
+   * @link https://react-query.tanstack.com/guides/query-invalidation
+   */
+  invalidate(
+    input?: unknown, // We can't infer the input type as it may be multiple queries!?
```

Comment:
Should be possible to do by grabbing the intersection of all the types & if there's an impossible one we can enforce ut to be `undefined`

Reply:
Done! 👍

Probably needs a bit more testing though ... next jobs!